src/main/java/org/terasology/fluid/system/FluidClientSystem.java

```diff
@@ -122,4 +130,45 @@ private void setFluidContainerIcon(EntityRef container, FluidContainerItemCompon
             }
         }
     }
+
+    /**
+     * Used to draw the Filling bar over the fluid container item in the cell.
+     *
+     * @param event  An event sent after the inventory cell has been rendered.
+     * @param entity The entity sending the request.
+     * @param fluidContainer FluidContainerItemComponent of the item.
+     */
+    @ReceiveEvent
+    public void drawFillingBarForFluidContainerItem(InventoryCellRendered event, EntityRef entity,
+                                                    FluidContainerItemComponent fluidContainer) {
+        Canvas canvas = event.getCanvas();
+
+        Vector2i size = canvas.size();
+
+        int minX = (int) (size.x * 0.8f);
+        int maxX = (int) (size.x * 0.9f);
+
+        int minY = (int) (size.y * 0.1f);
+        int maxY = (int) (size.y * 0.9f);
+
+        float fillingPercentage = 1f * fluidContainer.volume / fluidContainer.maxVolume;
```

Comment:
What is the `1f * ...` for? Both `volume` and `maxVolume` are floating point numbers, so the division should just work fine (see https://github.com/Terasology/Fluid/blob/develop/src/main/java/org/terasology/fluid/component/FluidContainerItemComponent.java#L33-L37)

```suggestion
        float fillingPercentage = fluidContainer.volume / fluidContainer.maxVolume;
```